### ISSUE TITLE
Old messages are run through new destinations on recovery

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3340]
This happens when no destinations (mirth_destinations) are set. On recovery, the engine has no way of knowing which destinations a message should be sent to; all it knows is that "null means all".

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3440
Reporter: narupley
Created: 2014-09-09T07:11:13.000-0700

**Comment by rbeckman-nextgen** [comment]
This should no longer be a problem because of MIRTH-3387, since the destinationSet variable will always be put in the source map. The only time where it's still a problem is if a user has unfinished messages and then upgrades to 3.1 in that state.

Imported Comment. Original Details:
Author: narupley
Created: 2014-09-09T07:13:46.000-0700

**Comment by rbeckman-nextgen** [comment]
Replicated this issue in 3.0.3. Created two test channels, one a default channel reader and the other a channel writer that sends to the previous channel; in the source transformer of the reader channel, I added a step to Thread.sleep(). After dispatching several messages, I halted the channel and then added new destinations. Upon redeploy and recovery, the messages were dispatched to all destinations.

I repeated these steps under 3.1 and only the destinations that existed at the initial time of dispatching received the message.

Imported Comment. Original Details:
Author: jaysenp
Created: 2014-09-11T10:22:25.000-0700
